server/src/main/java/org/elasticsearch/search/aggregations/metrics/MemoryTrackingTDigestArrays.java

```diff
@@ -193,39 +176,49 @@ public void resize(int newSize) {
         @Override
         public void ensureCapacity(int requiredCapacity) {
             if (requiredCapacity > array.length) {
-                int newSize = array.length + (array.length >> 1);
-                if (newSize < requiredCapacity) {
-                    newSize = requiredCapacity;
-                }
+                double[] oldArray = array;
+                // Used for used bytes assertion
+                long oldRamBytesUsed = ramBytesUsed();
+                long oldArraySize = RamUsageEstimator.sizeOf(oldArray);
+
+                int newSize = ArrayUtil.oversize(requiredCapacity, Double.BYTES);
+                int newArraySize = estimatedArraySize(newSize, Double.BYTES);
+                breaker.addWithoutBreaking(newArraySize);
```

Comment:
Shouldn't we be allowing tripping here, since the alloc follows next?

Reply:
Uh, indeed. I also forgot to update the resizes for the other arrays.
So, just fixed the non-breaking calls, and made all arrays have identical resizing logic. I couldn't abstract the repeated code tho, not at least without adding multiple abstract functions into the equation, as they work with primitive arrays